plugins/destination/s3/client/write.go

```diff
@@ -14,10 +15,24 @@ import (
 	"github.com/google/uuid"
 )
 
+const (
+	PathVarTable = "{{TABLE}}"
+	PathVarUUID  = "{{UUID}}"
+)
+
 func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data [][]any) error {
-	name := fmt.Sprintf("%s/%s.%s.%s", c.pluginSpec.Path, table.Name, c.pluginSpec.Format, uuid.NewString())
-	if c.pluginSpec.NoRotate {
-		name = fmt.Sprintf("%s/%s.%s", c.pluginSpec.Path, table.Name, c.pluginSpec.Format)
+	path := c.pluginSpec.Path
```

Comment:
We should update `c.pluginSpec.Path` during initialization with new spec, so that we don't repeat the same work over and over in `WriteTableBatch`.
`WriteTableBatch` should be just filling in the path with:
```go
name := strings.ReplaceAll(path, PathVarTable, table.Name)
name = strings.ReplaceAll(path, PathVarUUID, uuid.NewString()) // don't need to check as was checked during init phase
```

Reply:
Good suggestion, thanks! Updated :+1: